src/integer_mod_q/mat_polynomial_ring_zq/from.rs

```diff
@@ -32,66 +36,48 @@ impl<Mod: Into<ModulusPolynomialRingZq>> From<(&MatPolyOverZ, Mod)> for MatPolyn
     ///
     /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
     /// let poly_mat = MatPolyOverZ::from_str("[[4  -1 0 1 1, 1  42],[0, 2  1 2]]").unwrap();
-    /// let poly_ring_mat = MatPolynomialRingZq::from((&poly_mat, &modulus));
-    /// ```
-    fn from((matrix, modulus): (&MatPolyOverZ, Mod)) -> Self {
-        let mut out = Self {
-            matrix: matrix.clone(),
-            modulus: modulus.into(),
-        };
-        out.reduce();
-        out
-    }
-}
-
-impl<Mod: Into<ModulusPolynomialRingZq>> From<(MatPolyOverZ, Mod)> for MatPolynomialRingZq {
-    /// Creates a polynomial ring matrix of type [`MatPolynomialRingZq`] from
-    /// a [`MatPolyOverZ`] and a [`ModulusPolynomialRingZq`].
-    ///
-    /// Parameters:
-    /// - `matrix`: the polynomial matrix defining each entry.
-    /// - `modulus`: the modulus that is applied to each polynomial.
     ///
-    /// Returns a [`ModulusPolynomialRingZq`].
-    ///
-    /// # Examples
+    /// let poly_ring_mat = MatPolynomialRingZq::from((poly_mat, modulus));
     /// ```
-    /// use qfall_math::integer_mod_q::MatPolynomialRingZq;
-    /// use qfall_math::integer_mod_q::ModulusPolynomialRingZq;
-    /// use qfall_math::integer::MatPolyOverZ;
-    /// use std::str::FromStr;
-    ///
-    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
-    /// let poly_mat = MatPolyOverZ::from_str("[[4  -1 0 1 1, 1  42],[0, 2  1 2]]").unwrap();
-    /// let poly_ring_mat = MatPolynomialRingZq::from((poly_mat, &modulus));
-    /// ```
-    fn from((matrix, modulus): (MatPolyOverZ, Mod)) -> Self {
-        let mut out = MatPolynomialRingZq {
-            matrix,
+    fn from((matrix, modulus): (Matrix, Mod)) -> Self {
+        let mut out = Self {
+            matrix: matrix.into(),
             modulus: modulus.into(),
         };
         out.reduce();
         out
     }
 }
 
-impl From<&MatPolynomialRingZq> for MatPolynomialRingZq {
-    /// Alias for [`MatPolynomialRingZq::clone`].
-    fn from(value: &MatPolynomialRingZq) -> Self {
-        value.clone()
-    }
-}
-
```

Comment:
why do you remove this?

Reply:
This was a mistake, I did not want to remove it. I added it again in the next commit